### PR TITLE
Add playwright workflow

### PR DESCRIPTION
**Pull request by marthayaputra**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,6 +83,23 @@ Currently there is an issue where the above leads to react complaining about mul
 4. Open terminal in `glints-aries`.
 5. Run `yarn link react`.
 
+### Adding visual test
+
+We use `playwright` visual comparison test for components to check for regressions, since images in every browser and machine can be rendered differently we'll use docker image to generate and compare the screenshots.
```
Comment (ninariccimarie) [inline README.md]: can we include a link to playwright? this is to easily go to their documentation without having to search in a separate tab

File: README.md [hunk 2]
```diff
@@ -83,6 +83,23 @@ Currently there is an issue where the above leads to react complaining about mul
...
+
+All visual test is in `test/e2e` directory
+
+To generate new screenshots locally:
+
+Create docker image with `docker run --rm --network host -v $(pwd):/work/ -w /work/ -it mcr.microsoft.com/playwright:v1.29.0-focal /bin/bash`
+
+Run following commands in that image:
```
Comment (ninariccimarie) [inline README.md:96]: let's insert the command on each step so it's easier to follow

File: playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
+// eslint-disable-next-line prettier/prettier
+import type { PlaywrightTestConfig } from '@playwright/test';
+import { devices } from '@playwright/test';
+
+/**
+ * Read environment variables from file.
+ * https://github.com/motdotla/dotenv
+ */
+// require('dotenv').config();
```
Comment (ninariccimarie) [inline playwright.config.ts:9]: maybe clean up a bit on the commented out lines in this config file as well as if value we specify here is default

File: .github/workflows/cicd.yml [hunk 4]
```diff
@@ -22,17 +22,51 @@ jobs:
...
         run: yarn test
+
+  visual-test:
+    runs-on: ubuntu-latest
+    container:
+      image: mcr.microsoft.com/playwright:v1.29.0-focal
+      options: --user 1001
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (ninariccimarie) [inline .github/workflows/cicd.yml]: i think we can use v3?

File: .github/workflows/cicd.yml [hunk 5]
```diff
@@ -22,17 +22,51 @@ jobs:
...
       - name: Lint
         run: yarn lint
       - name: Test
         run: yarn test
+
+  visual-test:
+    runs-on: ubuntu-latest
+    container:
+      image: mcr.microsoft.com/playwright:v1.29.0-focal
+      options: --user 1001
```
Comment (ninariccimarie) [inline .github/workflows/cicd.yml:30]: is there a reason why we need to specify the user?

File: .github/workflows/cicd.yml [hunk 6]
```diff
@@ -22,17 +22,51 @@ jobs:
...
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v1
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Use Node.js
+        uses: actions/setup-node@v1
```
Comment (ninariccimarie) [inline .github/workflows/cicd.yml]: are we also able to use v3?

File: playwright.config.ts (hunk 3 above)
Comment (marthayaputra) [reply playwright.config.ts:9]: Hmm do you mean to remove all the comments? I think it's useful hint if you're new to playwright

File: .github/workflows/cicd.yml (hunk 5 above)
Comment (marthayaputra) [reply .github/workflows/cicd.yml:30]: this is for permission to write when we generate icons since it's in the docker env

File: README.md (hunk 1 above)
Comment (marthayaputra) [reply README.md]: included

File: README.md (hunk 2 above)
Comment (marthayaputra) [reply README.md:96]: updated

File: .github/workflows/cicd.yml (hunk 4 above)
Comment (marthayaputra) [reply .github/workflows/cicd.yml]: updated

File: .github/workflows/cicd.yml (hunk 6 above)
Comment (marthayaputra) [reply .github/workflows/cicd.yml]: updated

File: playwright.config.ts (hunk 3 above)
Comment (ninariccimarie) [reply playwright.config.ts:9]: just the other parts that we aren't using?

File: playwright.config.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    //   use: {
+    //     channel: 'msedge',
+    //   },
+    // },
+    // {
+    //   name: 'Google Chrome',
+    //   use: {
+    //     channel: 'chrome',
+    //   },
+    // },
```
Comment (ninariccimarie) [inline playwright.config.ts]: @marthayaputra like these lines? which can easily be found on the playwright documentation

File: README.md [hunk 8]
```diff
@@ -83,6 +83,23 @@ Currently there is an issue where the above leads to react complaining about mul
 4. Open terminal in `glints-aries`.
 5. Run `yarn link react`.
 
+### Adding visual test
+
+We use [playwright](https://playwright.dev/docs/test-snapshots) visual comparison test for components to check for regressions, since images in every browser and machine can be rendered differently we'll use docker image to generate and compare the screenshots.
+
+All visual test is in `test/e2e` directory
+
+To generate new screenshots locally:
```
Comment (ninariccimarie) [inline README.md:92]: maybe we can mention where the screenshots can be located?

File: playwright.config.ts [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  ],
+
+  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
```
Comment (ninariccimarie) [inline playwright.config.ts]: if we mention on our README where these are located, I think we can remove this

File: playwright.config.ts [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+  // outputDir: 'test-results/',
+
+  /* Run your local dev server before starting the tests */
+  // webServer: {
+  //   command: 'npm run start',
+  //   port: 3000,
+  // },
```
Comment (ninariccimarie) [inline playwright.config.ts]: if we don't need this, i think we should also remove this

File: playwright.config.ts (hunk 7 above)
Comment (marthayaputra) [reply playwright.config.ts]: updated

File: README.md (hunk 8 above)
Comment (marthayaputra) [reply README.md:92]: updated

**Comment by ninariccimarie** [comment]
LGTM 🚀

**Review by ninariccimarie — APPROVED** [review]

**Pull request merged**